### ISSUE TITLE
sql: properly report events for ownership changes

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#9585]
PR: https://github.com/cockroachdb/cockroach/pull/57969

From release notes:
> Both [`ALTER TABLE OWNER`](../v21.1/alter-table.html) and `REASSIGN OWNED BY` now report structured notable events about the ownership changes. Note that `REASSIGN OWNED BY` currently also reports an `alter_table_owner` for views and sequences that were implicitly reassigned, even though CockroachDB does not yet support the `ALTER VIEW OWNER` / `ALTER SEQUENCE OWNER` statements. [#57969][#57969] {% comment %}doc{% endcomment %}

**Comment by ericharmeling** [comment]
I believe this belongs to observability/security.

https://github.com/cockroachdb/cockroach/issues/57958

**Comment by taroface** [comment]
@piyush-singh Can you help assign and triage?

**Comment by mikeCRL** [comment]
@taroface I'm seeing `alter_table_owner` in the [generated docs](https://github.com/cockroachdb/cockroach/blob/release-21.1/docs/generated/eventlog.md), so I think as long as we link the release note for this to your new logging docs, we're good, and this issue can be closed. Can you confirm?

**Comment by taroface** [comment]
You're right that this should be covered by the logging docs, but I'm not sure if `REASSIGN OWNED BY`, which isn't yet documented, is covered in a separate release note. It looks like this SQL statement was added in https://www.cockroachlabs.com/docs/releases/v21.1.0-alpha.1.html

@ericharmeling @ianjevans Are you aware of another docs issue related to `REASSIGN OWNED BY`?

**Comment by ericharmeling** [comment]
> Are you aware of another docs issue related to `REASSIGN OWNED BY`?

I am not - it looks like it will come through on the next release note issue triage. Doesn't appear to be a roadmap priority though, so we can just through it on the backlog.

Also, FWIW, `ALTER VIEW OWNER` / `ALTER SEQUENCE OWNER` are now supported: https://github.com/cockroachdb/docs/pull/10396 (fixed https://github.com/cockroachdb/docs/issues/9614).
